### DISCUSSION TITLE
Apple Watch Battery Life

### DISCUSSION THREAD

**Discussion by HarmlessSaucer** [mKeRix/room-assistant#476] [General]
Hey!  
As a test I setup room-assistant on 3 Raspberry Pi’s:  2x ZeroW, 1x 3B+

I added my Apple Watch (Series 5) to it, but straight away I noticed a MASSIVE hit on battery life.  My Watch would normally last easy a day and a half with my usage, but now it was barely making it to 5PM.  

What are the things to look at on the room-assistant side?  Is it polling too much somehow?  Is this something I can tweak?  At the moment I’ve powered down the Pi’s as it just wasn’t viable.

**Comment by mKeRix** [comment]
That's interesting - I didn't see such a huge decline with my Series 2. You can try to increase the interval, that will make it be less aggressive:

```yaml
bluetoothClassic:
  interval: 20
  scanTimeLimit: 6
```

Additionally you could turn off scanning temporarily using the inquiries switches that should be available in Home Assistant, e.g. when you are sleeping.

**Reply by HarmlessSaucer** [reply]
Thanks for the details!  I totally missed the interval section somehow (doh- blame trying to do this with a toddler at home all day, sorry!).

Just to be sure it wasn’t my watch battery suddenly dying, I disconnected the Pi’s for a couple of days and my battery life suddenly massively improved.

With them connected I was barely getting to 5pm without getting the watch go into low power mode.
Disconnected them for a few days, and straight away it lasted the whole day, including workouts, then overnight doing sleep tracking and 9am next day it’s on 69%.  Radical difference.

I’ll tweak the settings and boot up the Pi’s again- I guess if something is wrong, I’m compounding the issue by having it duplicated on three Pis.

**Reply by mKeRix** [reply]
No worries! Generally adding more Pis into the game shouldn't make these issues worse, as room-assistant will use a round robin system to query the devices. So your watch will receive exactly 1 query each x (interval) seconds, no matter how many Pis you have setup with room-assistant running.

**Reply by Colorado4Wheeler** [reply]
This was good to know, I'll try that.  I noticed a massive Apple Watch battery hit as well but it didn't really "hit home" until my wife got in from out of town last weekend and said "ever since I got home my watch is dying twice as fast".  As a test I ran 24 hours with RA and logging battery level every hour and then 24 hours without RA and the difference is significant.  I'll try the interval and run another few test days to see where the sweet spot is.

**Reply by mKeRix** [reply]
Would love to hear about your results once you have them, maybe we can put some of that into the docs as reference for other people! :)

**Reply by Colorado4Wheeler** [reply]
I'm waiting for a full drain on my watch until I begin the interval testing, but currently the results are fairly staggering.  Without running all the numbers to get accurate statistics, and using a control of having my watch charged to 100% at 3:30 PM with the same activity, I'm seeing that from 3:30 PM to 8:00 AM with RA puts my watch at 18% while the same time frame without RA has my watch currently sitting at 65% with the exact same activity level for both days.

It is an interesting study, actually, as I eye other augmentations to my presence detection, such as facial recognition from my cameras, to see how various detection methods can be used in harmony to get the presence detection I desire with the smallest possible impact on the people I'm detecting.

I never thought that BT detection could have such an impact.  My initial assumption was that the watch was already broadcasting itself on the BT network so there would be no discernible battery impact as those packets were intercepted and acted upon, but I now realize that is not the case and that the polling is actually causing the watch to bring up the BT radio and drain the battery.

There was some hope that pure motion detection as my means of detecting presence would diminish significantly, but I can see now that there is a balance between them all - such as only having RA poll when there is motion in a room that is near an RA device (I've only got five Pi Zero's plus two Pi 4B's running RA right now) to just help complete the picture.

It's pretty fascinating stuff, I'm kind of geeking out on the "study" 😉.
